src/v1/function-builder.ts

```diff
@@ -192,6 +192,34 @@ function assertRuntimeOptionsValid(runtimeOptions: RuntimeOptions): boolean {
       );
     }
   }
+
+  if (
+    runtimeOptions.invoker !== undefined &&
+    typeof runtimeOptions.invoker === 'string' &&
+    runtimeOptions.invoker.length == 0
+  ) {
+    throw new Error(
+      'Invalid service account for function invoker, must be a non-empty string'
+    );
+  }
+  if (
+    runtimeOptions.invoker !== undefined &&
+    Array.isArray(runtimeOptions.invoker)
+  ) {
+    if (runtimeOptions.invoker.length == 0) {
+      throw new Error(
+        'Invalid invoker array, must contain at least 1 service account entry'
+      );
+    }
+    for (const serviceAccount of runtimeOptions.invoker) {
+      if (serviceAccount.length == 0) {
```

Comment:
Consider also checking that serviceAccount != 'public' or 'private'  here, since those don't really make sense in the array context.

Reply:
Added checks and updated unit tests